### PR TITLE
[ADO] Attestation code changes

### PR DESCRIPTION
**Pull request by arvindsingh-cloud**
## Description
</br>
 Brief description of the requested change.

## Checklist
- [ ] I have read the instructions mentioned in [Contribute to Code](/Contributing.md).
- [ ] I have read and understood the criteria described under [submitting changes](/Contributing.md#submitting-changes). 
- [ ] The title of the PR clearly describes the intent of the PR.
- [ ] This PR does not introduce any breaking changes to the code.

### CODE REVIEW

File: src/AzSK.AzureDevOps/Framework/Configurations/ADOAttestation.json [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    {
+    "refUpdates": [
+        {
+            "name": "refs/heads/master",
+            "oldObjectId": "{0}"
+        }
+    ],
+    "commits": [
+        {
+            "comment": "Added task markdown file.",
```
Comment (abhaydaga) [inline src/AzSK.AzureDevOps/Framework/Configurations/ADOAttestation.json:12]: need to change this.

File: src/AzSK.AzureDevOps/Framework/Configurations/ADOAttestation.json [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+"Delete": {
+    "refUpdates": [
+        {
+            "name": "refs/heads/master",
+            "oldObjectId": "{0}"
+        }
+    ],
+    "commits": [
+        {
+            "comment": "Deleted task markdown file.",
```
Comment (abhaydaga) [inline src/AzSK.AzureDevOps/Framework/Configurations/ADOAttestation.json:37]: need to change this.

File: src/AzSK.AzureDevOps/Framework/Configurations/ADOAttestation.json (hunk 1 above)
Comment (abhaydaga) [reply src/AzSK.AzureDevOps/Framework/Configurations/ADOAttestation.json:12]: Updated the attestation data.

File: src/AzSK.AzureDevOps/Framework/Configurations/ADOAttestation.json (hunk 2 above)
Comment (abhaydaga) [reply src/AzSK.AzureDevOps/Framework/Configurations/ADOAttestation.json:37]: Deleted the attestation data.

File: src/AzSK.AzureDevOps/Framework/Core/SVT/SVTControlAttestation.ps1 [hunk 3]
```diff
@@ -401,14 +409,24 @@ class SVTControlAttestation
...
-					if($this.controlStateExtension.GetControlStatePermission($resourceValue[0].FeatureName, $resourceValue[0].ResourceContext.ResourceName) )
+                    
+					if(($resourceValue[0].FeatureName -eq "Organization" -or $resourceValue[0].FeatureName -eq "Project") -and !$this.controlStateExtension.GetControlStatePermission($resourceValue[0].FeatureName, $resourceValue[0].ResourceContext.ResourceName) )
 					{
+					  Write-Host "Error: Attestation denied.`nThis may be because: `n  (a) You are attempting to attest controls for areas you do not have RBAC permission to.`n  (b) You are logged in using PAT Token (currently not supported for organization and project control's attestation)." -ForegroundColor Red
+					  continue
+					}
+					if($resourceValue[0].FeatureName -eq "Organization" -and !$this.controlStateExtension.GetProject())
+				    { 
+						Write-Host "`nNo project defined to store organization attestation details." -ForegroundColor Red
```
Comment (abhaydaga) [inline src/AzSK.AzureDevOps/Framework/Core/SVT/SVTControlAttestation.ps1:420]: No project defined to store attestation details for organization-specific controls.

File: src/AzSK.AzureDevOps/Framework/Core/SVT/SVTResourceResolver.ps1 [hunk 4]
```diff
@@ -289,12 +289,12 @@ class SVTResourceResolver: AzSKRoot
...
                                     $svtResource.ResourceType = "AzureDevOps.Build";
-                                    $svtResource.ResourceId = $bldDef.url
+                                    $svtResource.ResourceId = $bldDef.url.split('?')[0];
                                     $svtResource.ResourceTypeMapping = ([SVTMapping]::AzSKDevOpsResourceMapping |
                                                                     Where-Object { $_.ResourceType -eq $svtResource.ResourceType } |
                                                                     Select-Object -First 1)
-
-                                    $link = $svtResource.ResourceId.replace('_apis/build/Definitions','_build?definitionId=').split('?')[0];
+      
+                                    $link = $svtResource.ResourceId.replace('_apis/build/Definitions/','_build?definitionId=');
```
Comment (abhaydaga) [inline src/AzSK.AzureDevOps/Framework/Core/SVT/SVTResourceResolver.ps1:297]: do the same trimming for release objects

File: src/AzSK.AzureDevOps/Framework/Managers/ControlStateExtension.ps1 [hunk 5]
```diff
@@ -20,12 +20,17 @@ class ControlStateExtension
 	hidden [PSObject] $ControlSettings; 
 	hidden [PSObject] $resourceType;
 	hidden [PSObject] $resourceName;
+	hidden [PSObject] $resourceGroupName;
+	hidden [PSObject] $AttestationBody;
+	[bool] $IsPersistedControlStates = $false;
+	[bool] $IsExceptionChekingControlStateIndexerPresent = $false
```
Comment (abhaydaga) [inline src/AzSK.AzureDevOps/Framework/Managers/ControlStateExtension.ps1:26]: IsExceptionCheckingControlStateIndexerPresent

File: src/AzSK.AzureDevOps/Framework/Managers/ControlStateExtension.ps1 [hunk 6]
```diff
@@ -91,27 +96,25 @@ class ControlStateExtension
...
-					   $indexerObject =  $webRequestResult.value.value | ConvertFrom-Json;
-					   $loopValue = 0;
-					}
-					catch{
-						#Attestation index blob is not preset then return
-						$this.IsControlStateIndexerPresent = $false
-						return $true;
-					}
-					
+				  $this.IsExceptionChekingControlStateIndexerPresent = $false
```
Comment (abhaydaga) [inline src/AzSK.AzureDevOps/Framework/Managers/ControlStateExtension.ps1:99]: IsExceptionCheckingControlStateIndexerPresent

File: src/AzSK.AzureDevOps/Framework/Managers/ControlStateExtension.ps1 [hunk 7]
```diff
@@ -91,27 +96,25 @@ class ControlStateExtension
...
+				  $webRequestResult = $this.GetExtStorageContent( $this.IndexerBlobName );
+				  if($webRequestResult){
+				   $indexerObject = $webRequestResult 
+				  }
+				  else {
+					  if ($this.IsExceptionChekingControlStateIndexerPresent -eq $false) {
```
Comment (abhaydaga) [inline src/AzSK.AzureDevOps/Framework/Managers/ControlStateExtension.ps1:105]: IsExceptionCheckingControlStateIndexerPresent

File: src/AzSK.AzureDevOps/Framework/Managers/ControlStateExtension.ps1 [hunk 8]
```diff
@@ -250,79 +283,199 @@ class ControlStateExtension
...
 		   $webRequestResult = Invoke-RestMethod -Uri $uri -Method Get -ContentType "application/json" -Headers @{Authorization=("Basic {0}" -f $base64AuthInfo)}
-		   return $webRequestResult.value.value
+           if ($webRequestResult) {
+			return $webRequestResult
+		   }
+		   return $null;
 		}
 		catch{
+			if ($fileName -eq  $this.IndexerBlobName) {
+				$this.IsExceptionChekingControlStateIndexerPresent = $true
```
Comment (abhaydaga) [inline src/AzSK.AzureDevOps/Framework/Managers/ControlStateExtension.ps1:448]: IsExceptionCheckingControlStateIndexerPresent

File: src/AzSK.AzureDevOps/Framework/Managers/ControlStateExtension.ps1 [hunk 9]
```diff
@@ -142,7 +157,7 @@ class ControlStateExtension
 					$controlStateBlobName = $hashId + ".json"
 
 					$ControlStatesJson = $null;
-					$ControlStatesJson = $this.GetExtStorageContent($controlStateBlobName) | ConvertFrom-Json;
+					$ControlStatesJson = $this.GetExtStorageContent($controlStateBlobName)
```
Comment (abhaydaga) [inline src/AzSK.AzureDevOps/Framework/Managers/ControlStateExtension.ps1:160]: add relevant comments.

File: src/AzSK.AzureDevOps/Framework/Managers/ControlStateExtension.ps1 [hunk 10]
```diff
@@ -91,27 +96,25 @@ class ControlStateExtension
...
-					}
-					catch{
-						#Attestation index blob is not preset then return
-						$this.IsControlStateIndexerPresent = $false
-						return $true;
-					}
-					
+				  $this.IsExceptionChekingControlStateIndexerPresent = $false
+				  $webRequestResult = $this.GetExtStorageContent( $this.IndexerBlobName );
+				  if($webRequestResult){
```
Comment (abhaydaga) [inline src/AzSK.AzureDevOps/Framework/Managers/ControlStateExtension.ps1:101]: add comments.

Comment (arvindsingh-cloud) [reply src/AzSK.AzureDevOps/Framework/Managers/ControlStateExtension.ps1:103]: comment added

File: src/AzSK.AzureDevOps/Framework/Core/SVT/SVTResourceResolver.ps1 (hunk 4 above)
Comment (arvindsingh-cloud) [reply src/AzSK.AzureDevOps/Framework/Core/SVT/SVTResourceResolver.ps1:297]: versioning not comming in  release url

**Pull request merged**